appcli/orchestrators.py

```diff
@@ -492,7 +544,7 @@ def __compose_task(
 
     def __exec_command(self, command: str) -> CompletedProcess:
         logger.debug("Running [%s]", " ".join(command))
-        return subprocess.run(command)
+        return subprocess.run(command.split(" "), capture_output=True)
```

Comment:
tl;dr: Revert the function, change the `command` argument to be a list of string, and call this function with a list of string instead. If you need to add `capture_output=True`, then do so.

We shouldn't be changing a shared function unless necessary. Examining the internals of this function, we have now changed the external interface that this function provides. It now expects the `command` string to be able to be run with `subprocess.run` after doing a `.split(" ")` on it. This may potentially break functionality elsewhere in your class.

It looks like the `subprocess.run` function can take [a string, or a sequence of program arguments, but a sequence of arguments is generally preferred, as it allows the module to take care of any required escaping and quoting of arguments](https://docs.python.org/3/library/subprocess.html#frequently-used-arguments).

Given this, I think the `__exec_command` should specify that the `command` arg is a list of strings, and we shouldn't be doing any internal splitting. Doing splitting like this is dangerous, given the statement above.

It looks like we do pass in a list of strings for all other usages anyway, so we inadvertantly were using the `subprocess.run` function in the preferred manner. But since the [Python runtime does not enforce function and variable type annotations](https://docs.python.org/3/library/typing.html), this wasn't caught earlier.

Reply:
`.split(" ")` has been removed and the method signature has been changed to accept `Iterable[str]`